2_process/src/raster_per_PRMS.R

```diff
@@ -0,0 +1,94 @@
+rbind.fill.list <- function(x) {
+  nam <- sapply(x, names)
+  unam <- unique(unlist(nam))
+  len <- sapply(x, length)
+  out <- vector("list", length(len))
+  for (i in seq_along(len)) {
+    out[[i]] <- unname(x[[i]])[match(unam, nam[[i]])]
+  }
+  setNames(as.data.frame(do.call(rbind, out), stringsAsFactors=FALSE), unam)
+}
+
+raster_to_catchment_polygons <- function(polygon_sf, raster,
+                                         categorical_raster = NULL,
+                                         raster_summary_fun = NULL,
+                                         new_cols_prefix = NULL){
```

Comment:
Could be helpful to add `stop` conditions for raster summary being NULL when categorical is NULL or FALSE. Would this function work if the prefix is NULL? Seems like `paste` treats NULL as a space, so it's okay if it's left as NULL.

Reply:
Yea `paste()` treats `NULL` as nothing. So I planned to leave it in, the col name will simply have no prefix in `new_col_prefix` is `NULL`